datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -472,9 +472,6 @@ impl ExecutionPlan for RepartitionExec {
         if !self.maintains_input_order()[0] {
             result.clear_orderings();
         }
-        if self.preserve_order {
-            result = result.with_reorder(self.sort_exprs().unwrap_or_default().to_vec())
-        }
```

Comment:
For other reviewers: This is the key change that avoids losing alternative orderings.